### PR TITLE
[ycabled][active-active] no initialize Async Client, when no active-active cable type; fix names for all ycabled threads

### PR DESCRIPTION
**Pull request by vdahiya12**
This PR intends to not start sonic-ycabled's Asynchronous client, when there is no active-active cable type.
The Problem that we encounter is ycabled expects all threads to be working/running state all the time. 
Since there are no `active-active` cable_type Async client thread  exits gracefully, 
but by design this needs to be working if added to monitor loop thread. So this PR fixes this problem by runnig the Async Client only if configuration is `active-active` for atleast a cable. 

This PR also has all the infrastructure changes UT for all Classes having infinite loops in ycabled, thus helping in picking up cases if changes break the code. 

#### How Has This Been Tested?
Running on DUT and UT

### CODE REVIEW

**Review by prgeor — APPROVED** [review]
@vdahiya12 please fix the code coverage

**Review by zjswhhh — APPROVED** [review]
Please add ADO# and proper branch labels. 

Also please wait for approval from Prince G. before merging.

**Review by zjswhhh — APPROVED** [review]

**Comment by prgeor** [comment]
> Please add ADO# and proper branch labels.
> 
> Also please wait for approval from Prince G. before merging.

@yxieca please wait before merging to this repo

**Comment by prgeor** [comment]
@StormLiangMS please cherry pick to 202305

**Pull request merged**
